### PR TITLE
Jss plugin syntax rule value function

### PR DESCRIPTION
**Pull request by HenriBeck**
Tests inside the other packages which depend on the function plugin are currently broken. I don't see an easy way of fixing those tests though. Only when importing the other repositories.

### CODE REVIEW

File: packages/jss-plugin-syntax-rule-value-function/package.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "jss",
+    "style",
+    "sheet",
+    "stylesheet",
+    "css",
+    "components",
+    "composable",
+    "css in js",
+    "css-in-js",
+    "observable",
```
Comment (kof) [inline packages/jss-plugin-syntax-rule-value-function/package.json]: these last 3 are from observable, I think we can use: "dynamic", "function", "reactive", "animation"

File: packages/jss-plugin-syntax-rule-value-function/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "jss-plugin-syntax-rule-value-function",
+  "description": "JSS plugin for function syntax",
```
Comment (kof) [inline packages/jss-plugin-syntax-rule-value-function/package.json]: for function value and rule syntax

File: packages/jss-plugin-syntax-rule-value-function/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  "version": "0.1.0",
```
Comment (kof) [inline packages/jss-plugin-syntax-rule-value-function/package.json]: I guess it should correspond the jss version?

Comment (HenriBeck) [reply packages/jss-plugin-syntax-rule-value-function/package.json]: Lerna will take care of it when we publish them.

File: packages/jss-plugin-syntax-rule-value-function/src/integration.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable no-underscore-dangle */
```
Comment (kof) [inline packages/jss-plugin-syntax-rule-value-function/src/integration.test.js]: we can just call it index.test.js

File: packages/jss/src/Jss.js [hunk 5]
```diff
@@ -22,7 +21,7 @@ import type {
   generateClassName
 } from './types'
 
-const defaultPlugins = rulesPlugins.concat([functionsPlugin])
+const defaultPlugins = rulesPlugins
```
Comment (kof) [inline packages/jss/src/Jss.js]: we don't need defaultPlugins constant any more

File: packages/jss-plugin-syntax-rule-value-function/package.json (hunk 3 above)
Comment (HenriBeck) [reply packages/jss-plugin-syntax-rule-value-function/package.json]: https://github.com/lerna/lerna#publish

File: packages/jss/tests/integration/plugins.js [hunk 6]
```diff
@@ -387,7 +387,7 @@ describe('Integration: plugins', () => {
       })
       sheet = jss.createStyleSheet({
         a: {
-          color: data => data.color
```
Comment (kof) [inline packages/jss/tests/integration/plugins.js]: how is it possible that this test works? method .update triggers onUpdate only for functions ... hmmm

File: packages/jss-plugin-syntax-rule-value-function/.size-snapshot.json [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{
+  "dist/jss-observable.js": {
```
Comment (kof) [inline packages/jss-plugin-syntax-rule-value-function/.size-snapshot.json]: why do we have this one here?

File: packages/jss/tests/integration/plugins.js (hunk 6 above)
Comment (HenriBeck) [reply packages/jss/tests/integration/plugins.js]: onUpdate is being called for every plugin which implements onUpdate. That's why it works. It's not actually bound to the function plugin which is good.

Comment (kof) [reply packages/jss/tests/integration/plugins.js]: uhh, my bad

**Pull request merged**